src/freva/cli/history.py

```diff
@@ -77,26 +82,42 @@ def __init__(
             action="store_true",
             default=False,
         )
+        self.parser.add_argument(
+            "--json",
+            "-j",
+            help=(
+                "Display a json representation of the output, this can be"
+                "useful if you want to build shell based pipelines and want"
+                "parse the output with help of `jq`."
+            ),
```

Comment:
```
                "Display a json representation of the output, this can be"
                " useful if you want to build shell based pipelines and want to "
                "parse the output with the help of `jq`."
```


I run the `animator` plugin with this and I got:

```
$ freva-plugin animator variable=pr project=reanalysis product=reanalysis experiment=era5 institute=ecmwf model=afs time_frequency=1hr ensemble=r1i1p1 start=2017-08-02T00:00 end=2017-08-02T06:00 time_method=mean cmap=RdYlBu_r linecolor=k projection=PlateCarree proj_centre=50 pic_size=1360,900 plot_title=Precipitation cbar_label=true suffix=mp4 fps=8 --json
Setp 1: Collecting all files
Step 2: Opening the netcdf-files, collecting metadata
Traceback (most recent call last):
  File "/work/ch1187/clint/freva-dev/plugins/animator/run.py", line 506, in <module>
    cfg = main(parse_config(sys.argv))
  File "/work/ch1187/clint/freva-dev/plugins/animator/run.py", line 415, in main
    dset = xr.open_mfdataset(sorted(files), **mf_kwargs)
  File "/work/ch1187/clint/freva-dev/plugins/animator/plugin_env/lib/python3.10/site-packages/xarray/backends/api.py", line 947, in open_mfdataset
    raise OSError("no files to open")
OSError: no files to open
[15:11:18] ERROR    freva - ERROR - An error occured calling                                                                    plugin.py:328
                    ['/work/ch1187/clint/freva-dev/plugins/animator/plugin_env/bin/python', '-B',                                            
                    '/work/ch1187/clint/freva-dev/plugins/animator/run.py', '/tmp/tmp4nzb9c6d.json']                                         
           ERROR    freva - ERROR - Check also /work/ch1187/clint/freva-dev/share/slurm/animator/Animator-1172622.local         plugin.py:329
[15:11:19] ERROR    freva - ERROR - Command '['/work/ch1187/clint/freva-dev/plugins/animator/plugin_env/bin/python', '-B',        utils.py:85
                    '/work/ch1187/clint/freva-dev/plugins/animator/run.py', '/tmp/tmp4nzb9c6d.json']' returned non-zero exit                 
                    status 1. - increase verbosity flags (-v) for more information                                                           
                    ╭──────────────────────────────────── Traceback (most recent call last) ────────────────────────────────────╮            
                    │ /home/b/b380001/freva-dev/lib/python3.11/site-packages/evaluation_system/api/plugin.py:330 in _execute    │            
                    │                                                                                                           │            
                    │    327 │   │   │   if return_code and check:                                                              │            
                    │    328 │   │   │   │   log.error("An error occured calling %s", cmd)                                      │            
                    │    329 │   │   │   │   log.error("Check also %s", self.plugin_output_file)                                │            
                    │ ❱  330 │   │   │   │   raise sub.CalledProcessError(                                                      │            
                    │    331 │   │   │   │   │   return_code,                                                                   │            
                    │    332 │   │   │   │   │   cmd,                                                                           │            
                    │    333 │   │   │   │   )                                                                                  │            
                    ╰───────────────────────────────────────────────────────────────────────────────────────────────────────────╯            
                    CalledProcessError: Command '['/work/ch1187/clint/freva-dev/plugins/animator/plugin_env/bin/python', '-B',               
                    '/work/ch1187/clint/freva-dev/plugins/animator/run.py', '/tmp/tmp4nzb9c6d.json']' returned non-zero exit                 
                    status 1.                                
```

Reply:
```
$ freva-plugin psi project=cordex product=eur-11 institute=mpi-csc model=mpi-m-mpi-esm-lr-mpi-csc-remo2009-v1 experiment=historical time_frequency=day variable=pr ensemble=* cmor_table=day outputdir=/home/k/k204229/freva-dev/output/psi/20221208_133458 cache=/home/k/k204229/freva-dev/cache/psi cacheclear=true seldate=1961-01-01,1991-01-01 latlon=30,80,-20,30 area_weight=cos percentile_threshold=true threshold=99 persistence=3 sel_extr=10,5,3 dryrun=false --json
...
... program ended ...
[15:45:05] ERROR    freva - ERROR - (1146, "Table 'frevadb.history_output' doesn't exist") - increase verbosity flags (-v) for    utils.py:85
                    more information                                                                                                         
                    ╭──────────────────────────────────── Traceback (most recent call last) ────────────────────────────────────╮            
                    │ /home/b/b380001/freva-dev/lib/python3.11/site-packages/MySQLdb/connections.py:255 in query                │            
                    │                                                                                                           │            
                    │   252 │   │   # Since _mysql releases GIL while querying, we need immutable buffer.                       │            
                    │   253 │   │   if isinstance(query, bytearray):                                                            │            
                    │   254 │   │   │   query = bytes(query)                                                                    │            
                    │ ❱ 255 │   │   _mysql.connection.query(self, query)                                                        │            
                    │   256 │                                                                                                   │            
                    │   257 │   def _bytes_literal(self, bs):                                                                   │            
                    │   258 │   │   assert isinstance(bs, (bytes, bytearray))                                                   │            
                    ╰───────────────────────────────────────────────────────────────────────────────────────────────────────────╯            
                    ProgrammingError: (1146, "Table 'frevadb.history_output' doesn't exist")    
```
I do not now if this an issue with the db connection or there is sthing else going with the --json flag but I do not get it

Reply:
You'd have to use freva-dev fro this.

Reply:
I also have added a warning if you come from another freva and want to run the plugin command without batch mode. In batch mode I'll throw an error, because this is not working.